### PR TITLE
Optimize update logic & add a new strategy

### PR DESCRIPTION
**Pull request by imbajin**
1. if original data is not null but new data is null, use original data instead of null
2. add update strategy "OVERRIDE"

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/EdgeId.java [hunk 1]
```diff
@@ -193,7 +193,8 @@ public static EdgeId parse(String id) throws NotFoundException {
         String[] idParts = SplicingIdGenerator.split(id);
         if (!(idParts.length == 4 || idParts.length == 5)) {
             throw new NotFoundException("Edge id must be formatted as 4~5 parts"
-                                        + ", but got '%s'", id);
+                                        + ", but got %s parts,  '%s'",
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/id/EdgeId.java]: prefer move "parts" to the next line

**Comment by javeme** [comment]
there are conflicting files, please rebase.
and also update client.

File: hugegraph-api/src/main/java/com/baidu/hugegraph/define/UpdateStrategy.java [hunk 2]
```diff
@@ -139,6 +139,18 @@ Object updatePropertyValue(Object oldProperty, Object newProperty) {
         void checkPropertyType(Object oldProperty, Object newProperty) {
             this.checkCollectionType(oldProperty, newProperty);
         }
+    },
+
+    OVERRIDE {
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/define/UpdateStrategy.java]: upgrade api version

File: hugegraph-api/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java [hunk 3]
```diff
@@ -125,10 +124,10 @@ private LicenseParam initLicenseParam(LicenseVerifyParam param) {
     }
 
     private static LicenseVerifyParam buildVerifyParam(String path) {
-        InputStream stream = LicenseVerifier.class.getResourceAsStream(path);
         // NOTE: can't use JsonUtil due to it bind tinkerpop jackson
         ObjectMapper mapper = new ObjectMapper();
-        try {
+        try (InputStream stream =
```
Comment (Linary) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java]: good improve!

File: hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java [hunk 4]
```diff
@@ -90,13 +90,14 @@
      * [0.40] Issue-523: Add source_in_ring args for rings RESTful API
      * [0.41] Issue-493: Support batch updating properties by multiple strategy
      * [0.42] Issue-176: Let gremlin error response consistent with RESTful's
-     * [0.43] Issue-270 & 398: support shard-index and vertex + sortke prefix,
+     * [0.43] Issue-270 & 398: support shard-index and vertex + sortkey prefix,
      *        and split range to rangeInt, rangeFloat, rangeLong and rangeDouble
      * [0.44] Issue-633: Support unique index
+     * [0.45] Issue-673: Add a new update strategy
```
Comment (zhoney) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/version/ApiVersion.java]: make it clear:
```
Add OVERRIDE update strategy
```

**Pull request merged**